### ISSUE TITLE
xml parse functionality: problems in xml's Prolog and Document Type Declaration not reported

### ISSUE THREAD

**Issue by kaizer200** [prettydiff/prettydiff#203]
Hi,
I've found the following issues while using the 'parse' mode on xml document, relevant to the "Prolog and Document Type Declaration", using online editor running version: 1.15.17

**_1)**_
 **example:**

``` xml
<?xml version=1.0 encoding="utf-8"?>
 <tag1>
 </tag1>
```

**expected behaviour:**
 parse should report problem on first line, telling that attribute value 1.0, is missing quote  ("1.0"). 
**result:**
no error is issued.

**_2)**_
 **example:**

``` xml
<?xml version="1.0" encoding=utf-8"?>
 <tag1>
 </tag1>
```

**expected behaviour:**
- error is issued on the attribute value 'utf-8', indicating that its missing qoute (string not started...).
- several types of tokens are recognized (start, end, attribute, xml)

**result:**
-no error is issued.
-there is one token (type xml)

**Comment by prettydiff** [comment]
Currently the parse mode is only the raw parser without any analysis or intelligence.  The only intelligent things that do occur during parsing are things necessary to transform the parsed tokens for later use in the other modes.

These kinds of validations, analysis, and reporting are absolutely beneficial and I will add them eventually.  I will look at tackling these problems when I return my attention back to accessibility reporting: https://github.com/prettydiff/prettydiff/issues/112

One of the challenges I have not resolved with issue #112 is when to generate the reports and how to present the data.  Currently, parts of that issue are available, but only as supplemental meta-data from the beautification mode and only in the form of an HTML format.  This data needs to be available in an HTML format, plain text, and CLI output.  Your suggestion to integrate these reporting and analysis features into the parse mode, which at this time is hardly useful beyond minor debugging, is wise.

Accessibility reporting is a huge priority to me and part of accessibility is validating the markup.  I will return to these issues soon.

Here is my current agenda:

* Unit test automation.  I had been putting this off to the side for too long, and now I can no longer move forward without it.  I was attempting to add support for TypeScript, but this added support was breaking everything else.  My test samples, process, and code will be available for everybody to replicate thanks to NPM scripts.
* Coldfusion support.  This should be a quick win thanks to some recent guidance.  I will focus on this only after the unit test work is complete so that I can add CF code to my test samples.
* TypeScript support.  This is actually really hard.  TypeScript introduces some syntax features friendly to Java/C# that are otherwise foreign to JavaScript.  I am more than half way done with this.  Once I nail this I should be able to offer Java support with minor effort.
* Accessibility and this kind of code analysis.  I really want to take the time to get this area right, so I will get the others out of the way and really focus on these issues.

**Comment by prettydiff** [comment]
Pretty Diff 3 no longer performs analysis and now uses an external parser.  The parser autocorrects for unquoted attribute values by quoting them when they are mission.
